src/common/utils.ts

```diff
@@ -633,11 +691,186 @@ export const createFullfacetPipeline = <T, Y extends object>(
   facetObject["all"].push({
     $count: "totalSets",
   });
-  pipeline.push({ $facet: facetObject });
+  pipeline.push({
+    $facet: facetObject as Record<string, PipelineStage.FacetPipelineStage[]>,
+  });
 
-  return pipeline as PipelineStage[];
+  return pipeline;
 };
 
+// NOTE: remove below comments in the future once its sure that everything is
```

Comment:
I would drop this ( git  handles restoration for us ;) )

Reply:
Resolved!!!